### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by jaakkos**

### CODE REVIEW

**Comment by jaakkos** [comment]
Review and spell check plz.

File: src/clj_s3_client/core.clj [hunk 1]
```diff
@@ -73,17 +151,27 @@
...
+  [^AmazonS3Client client bucket-name object-key]
   (try
     (let [s3-object (.getObject client bucket-name object-key)
           s3-object-metadata (object-metadata->map (.getObjectMetadata s3-object))]
       (assoc s3-object-metadata :object-key (.getKey s3-object) :bucket-name (.getBucketName s3-object) :content (.getObjectContent s3-object)))
     (catch AmazonS3Exception e (suppress-not-found-exception e))))
 
-(defn delete-object [^AmazonS3Client client bucket-name object-key]
+(defn delete-object
+  "## (delete-object client \"my-awesome-bucket\" \"my-awsome-me.txt\")
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: my-awesome-me.txt

File: src/clj_s3_client/core.clj [hunk 2]
```diff
@@ -73,17 +151,27 @@
...
     (assoc result-s3-object-metadata :object-key object-key :bucket-name bucket-name :content is)))
 
-(defn- suppress-not-found-exception [exception]
-  (when-not (= 404 (.getStatusCode exception))
-    (throw exception)))
+(defn get-object
+  "## (get-object client \"my-awesome-bucket\" \"my-awsome-me.txt\")
 
-(defn get-object [^AmazonS3Client client bucket-name object-key]
+  Fetch object from given bucket with the key. Returns map which contains object metadata and contan as InputStream.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: contains an

Comment (mnylen) [reply src/clj_s3_client/core.clj]: or rather "which contains object metadata and an InputStream"

File: src/clj_s3_client/core.clj [hunk 3]
```diff
@@ -73,17 +151,27 @@
         result-s3-object-metadata (object-metadata->map (.getMetadata put-object-resp))]
     (assoc result-s3-object-metadata :object-key object-key :bucket-name bucket-name :content is)))
 
-(defn- suppress-not-found-exception [exception]
-  (when-not (= 404 (.getStatusCode exception))
-    (throw exception)))
+(defn get-object
+  "## (get-object client \"my-awesome-bucket\" \"my-awsome-me.txt\")
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: my-awesome-me.txt again

File: src/clj_s3_client/core.clj [hunk 4]
```diff
@@ -62,7 +44,103 @@
...
+  Checks if bucket exists.
+
+  Returns true or false
+  "
+  (.doesBucketExist client bucket-name))
+
+(defn put-object
+  "## (put-object client \"my-awesome-bucket\" \"my-awsome-me.txt\" file-input-stream {:acl :private :content-length 1234 :content-type \"text/html\"})
+
+  Store the input stream to s3 with give options.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: given options

File: src/clj_s3_client/core.clj [hunk 5]
```diff
@@ -62,7 +44,103 @@
...
+
+  An optional map of options can include any of the following keys:
+
+    :acl                    - sets accessing rights of the file supported values are
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: sets access rights of the file, supported values are:

File: src/clj_s3_client/core.clj [hunk 6]
```diff
@@ -62,7 +44,103 @@
...
+                                :private
+                                :public-read
+                                :public-read-write
+                                :aws-exec-read
+                                :authenticated-read
+                                :bucket-owner-read
+                                :bucket-owner-full-control
+                                :log-delivery-write
+
+    :content-length         - content lenght of InputStream.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: content _length_

File: src/clj_s3_client/core.clj [hunk 7]
```diff
@@ -62,7 +44,103 @@
...
+
+    :content-type           - the length of the content in bytes.
+
+    :content-language       - content language of the InputStream.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: should you specify the format here? iso two character languageccode?

File: src/clj_s3_client/core.clj [hunk 8]
```diff
@@ -62,7 +44,103 @@
...
+                                :public-read-write
+                                :aws-exec-read
+                                :authenticated-read
+                                :bucket-owner-read
+                                :bucket-owner-full-control
+                                :log-delivery-write
+
+    :content-length         - content lenght of InputStream.
+
+    :content-type           - the length of the content in bytes.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: content type? mime type? example would be nice (i.e. "image/png")

File: src/clj_s3_client/core.clj [hunk 9]
```diff
@@ -1,4 +1,8 @@
 (ns clj-s3-client.core
+  "Functions to interact with Amazon S3 storage.
+
+  All function takes as first parameter AmazonS3Client intance.
```
Comment (mnylen) [inline src/clj_s3_client/core.clj]: All functions take AmazonS3Client instance as first parameter.

File: src/clj_s3_client/core.clj [hunk 10]
```diff
@@ -62,7 +44,103 @@
...
+
+(defn delete-bucket
+  "## (delete-bucket client \"my-not-so-awesome-bucket\")
+
+  Delete bucket by given name.
+  "
+  [^AmazonS3Client client bucket-name]
+  (.deleteBucket client bucket-name))
+
+(defn exists?-bucket [^AmazonS3Client client bucket-name]
```
Comment (mnylen) [inline src/clj_s3_client/core.clj:101]: The function naming is a bit weird. `bucket-exists?` makes more sense. Feels somewhat out of place to have the question mark in the middle of function name.

**Pull request merged**